kingpin/actors/aws/ecs.py

```diff
@@ -406,7 +420,369 @@ def _execute(self):
+
+        Returns:
+            service_name to use.
+        """
+        # Optionally use specified 'service_name' instead of task family.
+        service_name = self.option('service_name')
+        if service_name is None:
+            service_name = family
+        return service_name
+
+    @gen.coroutine
+    def _describe_service(self, service_name):
+        """Describe a service by name.
+
+        Args:
+            service_name: service name to describe.
+
+        Returns:
+            Service param dict, or None if the service does not exist.
+        """
+        response = yield self.thread(
+            self.ecs_conn.describe_services,
+            cluster=self.option('cluster'),
+            services=[service_name])
+        self._handle_failures(response['failures'], self.FAILURE_MISSING)
+
+        services = response['services']
+
+        # There should never be more than one service for a given name.
+        if len(services) > 1:
+            self.log.warning(
+                'API returned more than one service for {name}'.format(
+                    name=service_name))
+
+        if services:
+            raise gen.Return(services[0])
+
+    @gen.coroutine
+    @dry('Would create service.')
+    def _create_service(self, service_name, family, revision, client_token):
+        """Create a service.
+
+        Args:
+            service_name: service name to use.
+            family: task definition family.
+            revision: task definition revision.
+            client_token: uuid for the service creation.
+        """
+        create_parameters = dict(
+            cluster=self.option('cluster'),
+            serviceName=service_name,
+            taskDefinition='{0}:{1}'.format(family, revision),
+            desiredCount=self.option('count'),
+            clientToken=client_token,
+            **self.service_definition)
+
+        self.log.info('Creating service.')
+
+        yield self.thread(
+            self.ecs_conn.create_service,
+            **create_parameters)
+
+    @gen.coroutine
+    @dry('Would update service.')
+    def _update_service(self, service_name, family, revision):
+        """Update a service.
+
+        Args:
+            service_name: service name to use.
+            family: task definition family.
+            revision: task definition revision.
+        """
+        deployment_configuration = self.service_definition[
+            'deploymentConfiguration']
+        update_parameters = dict(
+            cluster=self.option('cluster'),
+            service=service_name,
+            taskDefinition='{0}:{1}'.format(family, revision),
+            desiredCount=self.option('count'),
+            deploymentConfiguration=deployment_configuration)
+
+        self.log.info('Updating service.')
+
+        yield self.thread(
+            self.ecs_conn.update_service,
+            **update_parameters)
+
+    @gen.coroutine
+    def _register_service(self, service_name, family, revision):
```

Comment:
Would you mind renaming this to `_ensure_service()`? That follows the naming I've been using in the other actors that "ensure a resource is in a given state".